File: kafka-utils/src/main/java/com/wardziniak/kafka/scala/config/StreamsConfigBuilder.java
package com.wardziniak.kafka.scala.config;

import org.apache.kafka.streams.StreamsConfig;

import java.util.Optional;
import java.util.Properties;

public class StreamsConfigBuilder extends ConfigBuilder {

    private static final String DEFAULT_APPLICATION_ID = "applicationId1";

    private Optional<String> applicationId = Optional.empty();

    public StreamsConfigBuilder withApplicationId(String applicationId) {
        this.applicationId = Optional.of(applicationId);
        return this;
    }

    @Override
    public Properties buildConfig() {
        Properties properties = new Properties();
        properties.put(StreamsConfig.APPLICATION_ID_CONFIG, applicationId.orElse(DEFAULT_APPLICATION_ID));
        return properties;
    }
}

File: kafka-streams/src/main/scala/com/wardziniak/kafka/streams/scala/config/StreamsConfigBuilder.scala
package com.wardziniak.kafka.streams.scala.config

import java.util.Properties

import org.apache.kafka.common.serialization.Serdes
import org.apache.kafka.streams.StreamsConfig

case class StreamsConfigBuilder(
  bootstrapServer: String = "localhost:9092",
  applicationId: String = "applicationId1"
) {


  def buildConfig(): Properties = {
    val properties = new Properties()
    properties.put(StreamsConfig.APPLICATION_ID_CONFIG, applicationId)
    properties.put(StreamsConfig.BOOTSTRAP_SERVERS_CONFIG, bootstrapServer)
    properties.put(StreamsConfig.DEFAULT_KEY_SERDE_CLASS_CONFIG, Serdes.String().getClass)
    properties.put(StreamsConfig.DEFAULT_VALUE_SERDE_CLASS_CONFIG, Serdes.String().getClass)
    properties
  }
}
